tough/src/schema/key.rs

```diff
@@ -12,71 +14,113 @@ use std::collections::HashMap;
 use std::fmt;
 use std::str::FromStr;
 
+/// Serializes signing keys as defined by the TUF specification. All keys have the format
+/// ```json
+///  { "keytype" : "KEYTYPE",
+///     "scheme" : "SCHEME",
+///     "keyval" : "KEYVAL"
+///  }
+/// ```
+/// where:
+/// ```text
+///    KEYTYPE is a string denoting a public key signature system, such
+///    as RSA or ECDSA.
+///
+///    SCHEME is a string denoting a corresponding signature scheme.  For
+///    example: "rsassa-pss-sha256" and "ecdsa-sha2-nistp256".
+///
+///    KEYVAL is a dictionary containing the public portion of the key.
+/// ```
 #[derive(Debug, Clone, Deserialize, Serialize, PartialEq)]
 #[serde(rename_all = "kebab-case")]
 #[serde(tag = "keytype")]
 pub enum Key {
+    /// An RSA key.
     Rsa {
+        /// The RSA key.
         keyval: RsaKey,
+        /// The field denoting the key type as RSA.
         scheme: RsaScheme,
-
+        /// Any additional fields.
         #[serde(flatten)]
         _extra: HashMap<String, Value>,
     },
+    /// An Ed25519 key.
     Ed25519 {
+        /// The Ed25519 key.
         keyval: Ed25519Key,
+        /// The field denoting the key type as Ed25519.
         scheme: Ed25519Scheme,
-
+        /// Any additional fields.
         #[serde(flatten)]
         _extra: HashMap<String, Value>,
     },
+    /// An EcdsaKey
     Ecdsa {
+        /// The Ecdsa key.
         keyval: EcdsaKey,
+        /// The field denoting the key type as Ecdsa.
         scheme: EcdsaScheme,
-
+        /// Any additional fields.
         #[serde(flatten)]
         _extra: HashMap<String, Value>,
     },
 }
 
+/// Used to identify the type of RSA key scheme in use.
```

Comment:
```suggestion
/// Used to identify the RSA signature scheme in use.
```
I think that's the terminology?

Reply:
Yeah, that works better.